urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/gui/QuoteEntryDisplay.java

```diff
@@ -0,0 +1,282 @@
+                .addComponent(jLabel1)
+                .addComponent(buyPriceText, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
+                .addComponent(jLabel3)
+                .addComponent(sellPriceText, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
+                .addComponent(jButton1))
+                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
+                .addGroup(jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.CENTER)
+                .addComponent(jButton2)
+                .addComponent(sellVolumeText, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
+                .addComponent(jLabel4)
+                .addComponent(buyVolumeText, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
+                .addComponent(jLabel2))
+                .addGap(0, 0, Short.MAX_VALUE))
+                .addComponent(jSeparator2));
+
+        javax.swing.GroupLayout layout = new javax.swing.GroupLayout(getContentPane());
+        getContentPane().setLayout(layout);
+        layout.setHorizontalGroup(
+                layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
+                .addComponent(jPanel1, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE));
+        layout.setVerticalGroup(
+                layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
+                .addComponent(jPanel1, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE));
+
+        pack();
+    }// </editor-fold>//GEN-END:initComponents
+
+    /**
+     *
+     * @param p
+     */
+    public void setVisible(String p) {
+        product = p;
+        buyPriceText.setText("");
+        buyVolumeText.setText("");
+        sellVolumeText.setText("");
+        sellPriceText.setText("");
+        setTitle("Quote Entry for " + product);
+        super.setVisible(true);
+
+    }
+
+    private void jButton1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jButton1ActionPerformed
+
+        try {
+            if (buyPriceText.getText().trim().isEmpty()) {
+                throw new GUIDataException("Buy price cannot be empty");
+            }
+            if (sellPriceText.getText().trim().isEmpty()) {
+                throw new GUIDataException("Sell price cannot be empty");
+            }
+            if (buyVolumeText.getText().trim().isEmpty()) {
+                throw new GUIDataException("Buy Volume cannot be empty");
+            }
+            if (sellVolumeText.getText().trim().isEmpty()) {
+                throw new GUIDataException("Sell Volume cannot be empty");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [GUIDataException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [224].  Consider adding a catch clause of this specific exception in the outer try statement.